source/guides/put-it-together/ddiscordbot.md

```diff
@@ -0,0 +1,303 @@
+
+The `rows` argument is a list of lists. The main lists represent the separate rows, while the lists inside represent the components on each row (similarly to columns on a grid). This argument is found on both `discordinteraction` and `discordmessage`. You can make use of `<list_single>` and `.include_single` for this. Additionally, you can mix and match components in rows as you please. However, a message is still required!
+
+For example, sending a message with a button would look like:
+
+```dscript_blue
+buttons:
+    type: task
+    script:
+    - define click_me "<discord_button.with[id].as[click_me].with[label].as[Click me!].with[style].as[success]>"
+    - ~discordmessage id:mybot channel:(channel_id) rows:<list_single[<list_single[<[click_me]>]>]> Buttons!
+```
+
+The ID should be unique to the button: it's how you'll identify one button over another. You'll see this with the `discord button clicked` event, which has an `id` switch much like the slash command event's `name` switch. An example reply would look like:
+
+```dscript_green
+click_me:
+    type: world
+    events:
+        on discord button clicked id:click_me:
+        - ~discordinteraction defer interaction:<context.interaction>
+        - ~discordinteraction reply interaction:<context.interaction> "Hello, <context.interaction.user.name>!"
+```
+
+![](https://media.discordapp.net/attachments/533123622093455362/895896247641190420/unknown.png)
+
+Selection menus function in a similar way, although it's easier to see with an example script:
+
+```dscript_blue
+selection_menu:
+    type: task
+    script:
+    # Similar to command options
+    - definemap options:
+        1:
+            label: Good
+            value: mood_good
+            description: I'm doing well.
+            emoji: 🙂
+        2:
+            label: Better
+            value: mood_better
+            description: I'm doing wonderfully!
+            emoji: 😃
+        3:
+            label: Best
+            value: mood_best
+            description: I'm on top of the world!
+            emoji: 🤩
+
+    - define menu <discord_selection.with[id].as[mood_menu].with[options].as[<[options]>]>
+    - ~discordmessage id:mybot channel:(channel_id) rows:<list_single[<list_single[<[menu]>]>]> "How are you today?"
+
+mood_menu:
+    type: world
+    events:
+        on discord selection used id:mood_menu:
+        - ~discordinteraction defer interaction:<context.interaction>
+
+        - choose <context.option.get[value]>:
+            - case mood_good:
+                - define message "I'm glad."
+            - case mood_better:
+                - define message "Wow, that's great!"
+            - case mood_best:
+                - define message "Downright incredible!"
+
+        - ~discordinteraction reply interaction:<context.interaction> <[message]>
+```
+
+![](https://cdn.discordapp.com/attachments/533123622093455362/895890729530970132/eVkiFRrVa4.gif)
+
+### Some Things to Try
```

Comment:
Referencing other guide pages, eg: https://guide.denizenscript.com/guides/first-steps/ex-command.html
`to` should be titlecase
```suggestion
### Some Things To Try
```